libnavigation-core/src/test/java/com/mapbox/navigation/core/routealternatives/AlternativeRouteProgressDataProviderTest.kt

```diff
@@ -0,0 +1,289 @@
+                    listOf(List(23) { mockk() }),
+                    listOf(List(32) { mockk() }, List(19) { mockk() }),
+                    listOf(List(14) { mockk() }, List(23) { mockk() }), // 135 points for legs #0-#3
+                    listOf(List(8) { mockk() }),
+                )
+            )
+            val expected = RouteProgressData(4, 170, 35)
+
+            val actual = AlternativeRouteProgressDataProvider.getRouteProgressData(
+                primaryRouteProgressData,
+                alternativeMetadata
+            )
+
+            assertEquals(expected, actual)
+        }
+
+    @Test
+    fun `before fork, alternative starts after the primary route on the second leg`() =
+        coroutineRule.runBlockingTest {
+            val primaryRouteProgressData = RouteProgressData(4, 200, 35)
+            val alternativeMetadata = alternativeRouteMetadata(
+                primaryForkLegIndex = 5,
+                primaryForkRouteGeometryIndex = 250,
+                primaryForkLegGeometryIndex = 85,
+                alternativeForkLegIndex = 3,
+                alternativeForkRouteGeometryIndex = 220,
+                alternativeForkLegGeometryIndex = 81,
+                listOf(
+                    listOf(List(10) { mockk() }, List(7) { mockk() }, List(15) { mockk() }),
+                    listOf(List(23) { mockk() }), // 135 points for legs #0-#1
+                    listOf(List(32) { mockk() }, List(19) { mockk() }),
+                    listOf(List(14) { mockk() }, List(23) { mockk() }),
+                    listOf(List(8) { mockk() }),
+                )
+            )
+            val expected = RouteProgressData(2, 170, 119)
+
+            val actual = AlternativeRouteProgressDataProvider.getRouteProgressData(
+                primaryRouteProgressData,
+                alternativeMetadata
+            )
+
+            assertEquals(expected, actual)
+        }
+
+    @Test
+    fun `before fork, alternative starts after the primary route on the current leg`() =
+        coroutineRule.runBlockingTest {
+            // primary leg and route indices diff is 140
+            val primaryRouteProgressData = RouteProgressData(4, 200, 60)
+            val alternativeMetadata = alternativeRouteMetadata(
+                primaryForkLegIndex = 4,
+                primaryForkRouteGeometryIndex = 250,
+                primaryForkLegGeometryIndex = 110,
+                alternativeForkLegIndex = 0,
+                alternativeForkRouteGeometryIndex = 90,
+                alternativeForkLegGeometryIndex = 20,
+                listOf(
+                    listOf(List(10) { mockk() }, List(7) { mockk() }, List(15) { mockk() }),
+                    listOf(List(23) { mockk() }),
+                    listOf(List(32) { mockk() }, List(19) { mockk() }),
+                    listOf(List(14) { mockk() }, List(23) { mockk() }),
+                )
+            )
+            val expected = RouteProgressData(0, 40, 40)
```

Comment:
Again a weird test case. We return `RouteProgressData#legGeometryIndex = 40` while the fork is at index `alternativeForkLegGeometryIndex = 20`. So we'd be refreshing a portion _after_ the fork which should never happen.